bluemira/base/designer.py

```diff
@@ -55,18 +58,25 @@ def __init__(
         self,
         params: Union[Dict, ParameterFrame, ConfigParams, None],
         build_config: Optional[Dict] = None,
+        *,
+        verbose=True,
     ):
         self.params = make_parameter_frame(params, self.param_cls)
         self.build_config = build_config if build_config is not None else {}
-        self._run_func = self._get_run_func(self.run_mode)
+        self._verbose = verbose
 
     def execute(self) -> _DesignerReturnT:
         """
         Execute the designer with the run mode specified by the build config.
 
         By default the run mode is 'run'.
         """
-        return self._run_func()
+        return _timing(
+            self._get_run_func(self.run_mode),
```

Comment:
This change means that the run function is not defined until execute is called, but is does mean that the run function can get dynamic which is wasnt before, thoughts?

Reply:
This seems sensible to me. It means you can update the run mode in the build config and the change is honoured when you call `execute`.